src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -29,154 +29,121 @@ class TaskReplaceActor(
     val eventBus: EventStream,
     val readinessCheckExecutor: ReadinessCheckExecutor,
     val runSpec: RunSpec,
-    promise: Promise[Unit]) extends Actor with Stash with ReadinessBehavior with StrictLogging {
+    promise: Promise[Unit]) extends Actor with StrictLogging {
   import TaskReplaceActor._
 
   def deploymentId = status.plan.id
+  def pathId = runSpec.id
+
+  // instance to kill sorted by decreasing order of priority
+  // we always prefer to kill unhealthy tasks first
+  private[this] val toKillOrdered = instanceTracker.specInstancesSync(runSpec.id, readAfterWrite = true)
+    .partition(_.runSpecVersion == runSpec.version)._2.sortWith((i1, i2) => {
+      (i1.consideredHealthy, i2.consideredHealthy) match {
+        case (_, false) => false
+        case _ => true
+      }
+    })
 
-  // All existing instances of this app independent of the goal.
-  //
-  // Killed resident tasks are not expunged from the instances list. Ignore
-  // them. LaunchQueue takes care of launching instances against reservations
-  // first
-  val currentInstances = instanceTracker.specInstancesSync(runSpec.id, readAfterWrite = true)
-
-  // In case previous master was abdicated while the deployment was still running we might have
-  // already started some new tasks.
-  // All already started and active tasks are filtered while the rest is considered
-  private[this] val (instancesAlreadyStarted, oldInstances) = {
-    currentInstances.partition(_.runSpecVersion == runSpec.version)
-  }
+  // All instances to kill queued up
+  private[this] val toKill: mutable.Queue[Instance.Id] = toKillOrdered.map(_.instanceId).to[mutable.Queue]
 
-  // Old and new instances that have the Goal.Running
-  val activeInstances = currentInstances.filter(_.state.goal == Goal.Running)
+  private[this] var tick: Cancellable = null
 
-  // The ignition strategy for this run specification
-  private[this] val ignitionStrategy = computeRestartStrategy(runSpec, activeInstances.size)
+  @SuppressWarnings(Array("all")) // async/await
+  override def preStart(): Unit = {
+    super.preStart()
+    // subscribe to all needed events
+    eventBus.subscribe(self, classOf[HealthStatusResponse])
 
-  // compute all variables maintained in this actor =========================================================
+    // reset the launch queue delay
+    logger.info("Resetting the backoff delay before restarting the runSpec")
+    launchQueue.resetDelay(runSpec)
 
-  // Only old instances that still have the Goal.Running
-  val oldActiveInstances = oldInstances.filter(_.state.goal == Goal.Running)
+    // Fetch state of apps
+    val system = akka.actor.ActorSystem("system")
+    tick = system.scheduler.schedule(0 seconds, 10 seconds)(request_health_status)
```

Comment:
this hardcoded value will likely be frowned upon by upstream. I agree it's a good thing at first but we should add a comment to make it changed (configurable?) in the future